src/WorkspaceHelper.php

```diff
@@ -234,6 +237,14 @@ public static function checkPermission($element, $type)
             return true;
         }
 
+        $event = new PermissionEvent($element, $type);
+        /** @var EventDispatcher $eventDispatcher */
+        $eventDispatcher = \Pimcore::getContainer()->get('event_dispatcher');
+        $eventDispatcher->dispatch($event, PermissionEvents::PRE_CHECK);
+        if (!$event->isGranted()) {
+            throw new ClientSafeException('access for '.  $element->getFullPath() . ' denied');
```

Comment:
IMHO, it only should throw an expection if the policy is NOT_ALLOWED_POLICY_EXCEPTION. see two lines below. wdyt?

Reply:
yes you're right